generator/src/generator/namespace.rs

```diff
@@ -590,6 +617,20 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                 name = name,
             ));
         }
+
+        if gathered.needs_lifetime {
+            enum_cases.request_into_owned_cases.push(format!(
+                "Request::{ns_prefix}{name}(req) => Request::{ns_prefix}{name}(req.into_owned()),",
+                ns_prefix = ns_prefix,
+                name = name,
+            ));
+        } else {
+            enum_cases.request_into_owned_cases.push(format!(
+                "Request::{ns_prefix}{name}(req) => Request::{ns_prefix}{name}(req),",
+                ns_prefix = ns_prefix,
+                name = name,
```

Comment:
Random idea (doesn't have to be in this PR): Can this be replaced with a simple `e => e` at the end of a match instead of all this individual lines?
I guess that this could not work due to lifetimes, but if I am wrong about this and it does work, then that would get rid of quite a number of lines of code.

Reply:
Unfortunately that doesn't work because it does trigger lifetime issues.

```
error[E0495]: cannot infer an appropriate lifetime due to conflicting requirements
    --> src/protocol/mod.rs:3302:15
     |
3302 |         match self {
     |               ^^^^
     |
note: first, the lifetime cannot outlive the lifetime `'input` as defined on the impl at 1260:6...
    --> src/protocol/mod.rs:1260:6
     |
1260 | impl<'input> Request<'input> {
     |      ^^^^^^
note: ...so that the types are compatible
    --> src/protocol/mod.rs:3302:15
     |
3302 |         match self {
     |               ^^^^
     = note: expected  `protocol::Request<'_>`
                found  `protocol::Request<'input>`
     = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the expression is assignable
    --> src/protocol/mod.rs:3302:9
     |
3302 | /         match self {
3303 | |             Request::Unknown(header, body) => Request::Unknown(header, Cow::Owned(body.into_owned())),
3304 | |             Request::CreateWindow(req) => Request::CreateWindow(req.into_owned()),
3305 | |             Request::ChangeWindowAttributes(req) => Request::ChangeWindowAttributes(req.into_owned()),
...    |
3561 | |             passthrough_case => passthrough_case,
3562 | |         }
     | |_________^
     = note: expected  `protocol::Request<'static>`
                found  `protocol::Request<'_>`
```

The compiler should be smarter than that but it isn't.